trino-parser/src/main/java/io/trino/sql/ExpressionFormatter.java

```diff
@@ -998,61 +996,9 @@ private String processDateDiffInBigQuery(FunctionCall node, Void context)
         }
     }
 
-    static String formatStringLiteral(String s, Dialect dialect)
+    static String formatStringLiteral(String s)
     {
-        if (dialect == BIGQUERY) {
-            s = s.replace("'", "\\'");
-        }
-        else {
-            s = s.replace("'", "''");
-        }
-        if (CharMatcher.inRange((char) 0x20, (char) 0x7E).matchesAllOf(s)) {
-            return "'" + s + "'";
-        }
-
-        StringBuilder builder = new StringBuilder();
-        if (dialect == DEFAULT || dialect == DUCKDB || dialect == POSTGRES) {
```

Comment:
I saw there're some DuckDB-specific behaviors. Do they work well if we remove the dialect?

Reply:
Sounds great.